project/SparkBuild.scala

```diff
@@ -756,10 +757,12 @@ object SparkConnect {
     // `netty-*.jar` and `unused-1.0.0.jar` from assembly.
     (assembly / assemblyExcludedJars) := {
       val cp = (assembly / fullClasspath).value
-      cp filter { v =>
-        val name = v.data.getName
-        name.startsWith("pmml-model-") || name.startsWith("scala-collection-compat_") ||
-          name.startsWith("jsr305-") || name.startsWith("netty-") || name == "unused-1.0.0.jar"
+      val validPrefixes = Set("spark-connect", "unused-", "guava-", "failureaccess-",
```

Comment:
If this modification is not made, the connect server assembly jar will not be able to start after upgrade arrow 15 because too many unnecessary jar files are included in the assembly jar, which has caused some conflicts after upgrading to Arrow 15.

But I think this change is quite universal, so I will submit this change in a separate pr first.

Reply:
+1 for that.

Reply:
https://github.com/apache/spark/pull/44798